doc/CommonTraps.md

```diff
@@ -0,0 +1,32 @@
+# Common PowerPoint Add-in Development Traps
+
+This document details the common pitfalls when developing for PowerPointLabs, so that these mistakes can be noted/avoided by future contributors.
+
+1. The index of an Office collection starts from 1, however note that the index of a collection provided by .NET starts from 0
+`ShapeRange range = getCurrentSelectedShapes(); range[1] ... ; //the first element`
+
+1. When you delete, cut, or ungroup a shape programmatically, and you undo this change from the PowerPoint UI, then that shape will become corrupted. For a corrupted shape, accessing attributes programatically other than it's `left`/`top`/`width`/`height`/`name` would throw an exception. <br><br>One possible solution would be re-allocate the shape’s memory; copy and paste it programmatically before you handle the shape
```

Comment:
Could mention ShapeUtil has a method called CorruptionCorrection to fix this corruption of shapes.